gordon-plugin/src/main/kotlin/com/banno/gordon/GordonTestTask.kt

```diff
@@ -179,16 +179,21 @@ internal abstract class GordonTestTask @Inject constructor(
                 signingConfig = signingConfig,
                 instrumentationApk = instrumentationApk,
                 installTimeoutMillis = installTimeoutMillis.get(),
-                adb = adb
+                adb = adb,
+                ignoreProblematicDevices = ignoreProblematicDevices.get(),
+                problematicDevices = problematicDevices,
             ).bind()
 
+            pools = pools.filterProblematicDevices(problematicDevices)
+            pools.validateDevicePools().bind()
+
             val testResults = runAllTests(
                 dispatcher = Dispatchers.Default,
                 logger = logger,
                 instrumentationPackage = instrumentationPackage.get(),
                 instrumentationRunnerOptions = instrumentationRunnerOptions.get(),
                 allTestCases = testCases,
-                allPools = pools,
+                allPools = pools.filterProblematicDevices(problematicDevices),
```

Comment:
`pools` would already be filtered at this point, so we don't need to do it again, do we?

```suggestion
                allPools = pools,
```